Sources/Xcodeproj/pbxproj().swift

```diff
@@ -87,33 +87,94 @@ public func pbxproj(srcroot: AbsolutePath, projectRoot: AbsolutePath, xcodeprojP
     for module in modules {
         // Base directory for source files belonging to the module.
         let moduleRoot = module.sources.root
-        
+
         // Contruct an array of (refId, path, bflId) tuples for all the source files in the model.  The reference id is for the PBXFileReference in the group hierarchy, and the build file id is for the PBXBuildFile in the CompileSources build phase.
         let sourceFileRefs = fileRefs(forModuleSources: module, srcroot: srcroot)
-        
-        // Make an array of all the source file reference ids to add to the main group.
-        let sourceRefIds = sourceFileRefs.map{ $0.refId }
 
-        // the “Project Navigator” group for this module
-        print("        \(module.groupReference) = {")
-        print("            isa = PBXGroup;")
-        print("            name = '\(module.name)';")
-        print("            path = '\(moduleRoot.relative(to: projectRoot).asString)';")
-        print("            sourceTree = '<group>';")
-        print("            children = (" + sourceRefIds.joined(separator: ", ") + ");")
-        print("        };")
+        // Hash of AbsolutePath of a group and reference id's of its children.
+        // Children can be either a source file or a nested group
+        var groupsChildren = [AbsolutePath: Set<String>]()
+
+        // reference id's of immediate children of this module group.
+        var topLevelRefs  = Set<String>()
 
         // the contents of the “Project Navigator” group for this module
         for fileRef in sourceFileRefs {
             let path = fileRef.path.relative(to: moduleRoot)
             print("        \(fileRef.refId) = {")
+            print("            name = \(fileRef.path.basename);")
             print("            isa = PBXFileReference;")
             print("            lastKnownFileType = \(module.fileType(forSource: path));")
-            print("            path = '\(fileRef.path.relative(to: moduleRoot).asString)';")
+            print("            path = '\(fileRef.path.relative(to: srcroot).asString)';")
```

Comment:
I understand why this has to change to srcroot-relative, given that the groups that are created don't have paths assigned, and it's ok for now.  A cleaner approach would be for every group and fileref to have a `path` consisting of just its name, letting the `sourceTree = '<group>'` part provide the chaining up to the root group level.  Then the `name` property can also be omitted.  But this can be done as a future improvement, in the interest of landing this PR.  The current diffs aren't wrong, just a bit less elegant.